File: Qb.Net/SqlOm/OmUnion.cs
using System.Collections;
using System.Collections.Generic;

namespace Viten.QueryBuilder.SqlOm
{

  /// <summary></summary>
  public class OmUnionItem
  {
    /// <summary></summary>
    public SelectQuery Query { get; set; }


    /// <summary></summary>
    public UnionMod RepeatingAction { get; set; }
    /// <summary>
    /// 
    /// </summary>
    public OmUnionItem()
    {
    }

    /// <summary></summary>
    public OmUnionItem(SelectQuery query, UnionMod repeatingAction)
    {
      Query = query;
      RepeatingAction = repeatingAction;
    }

  }

  /// <summary>
  /// Encapsulates SQL UNION statement
  /// </summary>
  public class OmUnion
  {
    List<OmUnionItem> items = new List<OmUnionItem>(5);
    /// <summary>
    /// Creates a new SqlUnion
    /// </summary>
    public OmUnion()
    {
    }

    /// <summary>
    /// Adds a query to the UNION clause
    /// </summary>
    /// <param name="query">SelectQuery to be added</param>
    /// <remarks>Query will be added with DistinctModifier.Distinct </remarks>
    public void Add(SelectQuery query)
    {
      Add(query, UnionMod.Distinct);
    }

    /// <summary>
    /// Adds a query to the UNION clause with the specified DistinctModifier
    /// </summary>
    /// <param name="query">SelectQuery to be added</param>
    /// <param name="repeatingAction">Distinct modifier</param>
    public void Add(SelectQuery query, UnionMod repeatingAction)
    {
      items.Add(new OmUnionItem(query, repeatingAction));
    }

    /// <summary></summary>
    public IList Items
    {
      get { return items; }
    }

  }
}

File: Qb.Js/src/OmUnion.ts
import { SelectQuery } from "./SelectQuery"
import { UnionMod } from "./Enums"

// interface IOmUnionItem {
//     Query: SelectQuery;
//     RepeatingAction: DistinctModifier;
//   }

export class OmUnionItem /*implements IOmUnionItem*/ {
  Query: SelectQuery;
  RepeatingAction: UnionMod;

  setQuery(query: SelectQuery) {
    this.Query = query;
  }


  setRepeatingAction(repeatingAction: UnionMod) {
    this.RepeatingAction = repeatingAction;
  }

  constructor(query: SelectQuery, repeatingAction: UnionMod) {
    this.Query = query;
    this.RepeatingAction = repeatingAction;
  }
}

// interface IOmUnion {
//   add(query: SelectQuery, repeatingAction: DistinctModifier);
//   Items: Array<OmUnionItem>;
// }
export class OmUnion /*implements IOmUnion*/ {
  Items = new Array<OmUnionItem>();
  add(query: SelectQuery, repeatingAction: UnionMod) {
    this.Items.push(new OmUnionItem(query, repeatingAction));
  }
}
